apps/backend/src/data-marts/entities/insight.entity.ts

```diff
@@ -25,6 +27,19 @@ export class Insight {
   @JoinColumn()
   dataMart: DataMart;
 
+  @Column({ nullable: true })
+  lastDataMartRunId?: string;
+
+  @OneToOne(() => DataMartRun)
+  @JoinColumn({ name: 'lastDataMartRunId' })
+  lastDataMartRun?: DataMartRun | null;
```

Comment:
I would recommend decouple these entities. Foreign keys can cause unexpected trouble when we decide to clean the DataMartRun table. Besides, we already have the lastDataMartRunId and can fetch an entity when and if we need it.